drivers/video/gc2145.c

```diff
@@ -1032,14 +1032,24 @@ static int gc2145_set_fmt(const struct device *dev, enum video_endpoint_id ep,
 		return -ENOTSUP;
 	}
 
-	width = fmt->width;
-	height = fmt->height;
-
 	if (memcmp(&drv_data->fmt, fmt, sizeof(drv_data->fmt)) == 0) {
 		/* nothing to do */
 		return 0;
 	}
 
+	/* Check if camera is capable of handling given format */
+	for (int i = 0;; i++) {
+		if (i == ARRAY_SIZE(fmts)) {
+			LOG_ERR("Image format not supported\n");
+			return -ENOTSUP;
+		}
+		if (fmts[i].width_min == fmt->width && fmts[i].height_min == fmt->height &&
+		    fmts[i].pixelformat == fmt->pixelformat) {
+			res = (enum resolutions)i;
+			break;
+		}
+	}
```

Comment:
Nit: I think a more common way to write this would be:
```C
	enum resolutions res = RESOLUTIONS_MAX;
	
	for (int i = 0; i < ARRAY_SIZE(fmts); i++) {
		if (fmts[i].width_min == fmt->width && fmts[i].height_min == fmt->height &&
		    fmts[i].pixelformat == fmt->pixelformat) {
			res = (enum resolutions)i;
			break;
		}
	}
	if (res == RESOLUTIONS_MAX) {
		LOG_ERR("Image format not supported\n");
		return -ENOTSUP;
	}
```

Reply:
Then there is the more obvious `for (i = 0; i < N; i++)` idiom standing out.
I appreciate the teachings. :)